spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -110,6 +110,10 @@ public void setHeaderMapper(HeaderMapper<Map<String, String>> headerMapper) {
     this.headerMapper = headerMapper;
   }
 
+  public HeaderMapper<Map<String, String>> getHeaderMapper() {
+    return this.headerMapper;
+  }
+
```

Comment:
Aha! I thought it was odd that the `headerMapper` property ended up being mutable!
Generally, returning an object that the caller can then change is a dangerous practice because then this class (`PubSubInboundChannelAdapter`) can't guarantee the integrity of its own state. 
In this case, you are explicitly returning it in order to change it -- creating a great example of why this kind of "getter of a mutable object/collection" pattern is dangerous.

Please remove the getters here and in the `PubSubMessageHandler`, and use the existing setter instead.